### PR TITLE
Use maxRedirects only for Redirect errors and maxRetries for network errors

### PR DESCRIPTION
**Pull request by duke-cliff**
Use maxRedirects only for Redirect errors(moved/ask/read only/loading...)
use maxRetries for network error retries.

### CODE REVIEW

File: cluster.go [hunk 1]
```diff
@@ -766,9 +768,15 @@ func (c *ClusterClient) _process(ctx context.Context, cmd Cmder) error {
 	var node *clusterNode
 	var ask bool
 	var lastErr error
-	for attempt := 0; attempt <= c.opt.MaxRedirects; attempt++ {
-		if attempt > 0 {
-			if err := internal.Sleep(ctx, c.retryBackoff(attempt)); err != nil {
+	var failureAttempt = 0
+	var redirectAttempt = 0
+	for redirectAttempt <= c.opt.MaxRedirects || failureAttempt < c.opt.MaxRetries {
```
Comment (vmihailenco) [inline cluster.go:773]: You probably want it to be `&&`, not `||`.

**Comment by vmihailenco** [comment]
This only updates part of the code - pipelines still use `MaxRedirects`.

**Comment by vmihailenco** [comment]
Closing since this is rather complicated and there is no progress.

**Comment by shenqianjin** [comment]
this change looks good.

@vmihailenco why did we close it? is there any alternative solution about this ?

**Pull request closed without merging**